### PR TITLE
Added Todo List Feature

### LINKED ISSUE #255: Add To Do List
Extension is preety Good, but if you add 1 more option that is to add anyone can add their To Do List and mark as done or not done. And To Do list should reset daily after 23:59.

### PR DESCRIPTION
**Pull request by Thunder-Blaze**
## Description
Added a ToDo List on the right corner of the page.
- Followed the default colorscheme.
- The todo list has an input box to add items.
- You can either click the plus button or press enter to add an item
- It alerts the user to write something if the input field is empty
- And each added item has a remove button to remove them from the list.
- You can check the list item by clicking on it.
- The list will automatically reset after 23:59

## Related Issues
- Closes #255 

## Checklist
- [x] I have read and followed the [Contributing Guidelines](../CONTRIBUTING.md).
- [x] The ToDo List refreshes next day automatically
- [x] I have done necessary testing.
- [x] Added a toggle for Todo List in Settings Page

https://github.com/user-attachments/assets/bf5161ed-634c-4148-9615-50473c9a88a0

https://github.com/user-attachments/assets/2281c9f9-97d3-464c-a18c-26c96050b1c0

### CODE REVIEW

**Comment by prem-k-r** [comment]
Great work, really impressive!
It would be even better if you could add a toggle to hide or show the ToDo list in the menu.

**Comment by Thunder-Blaze** [comment]
Done :D 
there is only one Issue, the To Do List auto closes (as it looses focus) when a item is removed from it

**Comment by prem-k-r** [comment]
Btw guys, on an unrelated note, I noticed that the menu icon is not perfectly centered in the screen recording.

**Comment by Thunder-Blaze** [comment]
hmm, can you clarify which menu icon are we talking about

**Comment by prem-k-r** [comment]
> hmm, can you clarify which menu icon are we talking about

image
As I said, it's unrelated to the changes made in the PR

**Comment by prem-k-r** [comment]
I can only add new items in list.. unable to delete or complete it
error:

Refused to execute inline event handler because it violates the following Content Security Policy directive: "script-src 'self'". Either the 'unsafe-inline' keyword, a hash ('sha256-...'), or a nonce ('nonce-...') is required to enable inline execution. Note that hashes do not apply to event handlers, style attributes and javascript: navigations unless the 'unsafe-hashes' keyword is present.
Understand this errorAI
index.html:1337 Refused to execute inline event handler because it violates the following Content Security Policy directive: "script-src 'self' 'wasm-unsafe-eval' 'inline-speculation-rules' http://localhost:* http://127.0.0.1:*". Either the 'unsafe-inline' keyword, a hash ('sha256-...'), or a nonce ('nonce-...') is required to enable inline execution. Note that hashes do not apply to event handlers, style attributes and javascript: navigations unless the 'unsafe-hashes' keyword is present.

**Comment by prem-k-r** [comment]
in windows/chrome

File: script.js [hunk 1]
```diff
@@ -278,6 +278,93 @@ document.addEventListener("click", function (event) {
...
+    if (event.target.tagName == "LI"){
+        event.target.classList.toggle("checked");
+        SaveData();
+    } else if (event.target.tagName == "SPAN"){
+        event.target.parentElement.remove();
+        SaveData();
+    }
+}
+function SaveData(){
+    localStorage.setItem("todolist", todoulList.innerHTML)
```
Comment (prem-k-r) [inline script.js:324]: please no direct use of innerHTML, to prevent XSS vulnerabilities.

**Review by prem-k-r — CHANGES_REQUESTED** [review]

File: script.js (hunk 1 above)
Comment (Thunder-Blaze) [reply script.js:324]: can I use innerText ?

Comment (prem-k-r) [reply script.js:324]: yep

**Comment by prem-k-r** [comment]
There is one minor issue..  upon page reload, todo icon briefly flash in default blue color theme for a split second.
this is also for google apps 9-dot icon.
check https://github.com/XengShi/materialYouNewTab/commit/6fa436aa7d7e677ced20d260a3115ce1aa1cabb8 , same approach can be used

Also can you please make alert translatable like other alerts

No hurry bro.. take your time

File: languages.js [hunk 2]
```diff
@@ -867,6 +879,7 @@ const translations = {
         "failedbackup": "ব্যাকআপ ব্যর্থ: ",
         "restorecompleted": "পুনরুদ্ধার সফলভাবে সম্পন্ন হয়েছে!",
         "restorefailed": "পুনরুদ্ধার ব্যর্থ: ",
+        "emptytodoinput": "ইউ মাস্ট রাইট সামথিং",
```
Comment (prem-k-r) [inline languages.js:882]: Wrong translation

File: languages.js [hunk 3]
```diff
@@ -1670,8 +1691,8 @@ const translations = {
         "imagedimensions": "तयार केलेली इमेज ({width}x{height}) चे परिमाण 1920x1080 पिक्सल्सपेक्षा जास्त आहे. \n\nयामुळे कार्यक्षमता प्रभावित होऊ शकते किंवा इमेज नीट लोड होऊ नये.",
         "failedbackup": "बॅकअप अयशस्वी: ",
         "restorecompleted": "पुनर्संचयित यशस्वी झाले!",
-        "restorefailed": "पुनर्संचयित अयशस्वी: "
-
+        "restorefailed": "पुनर्संचयित अयशस्वी: ",
+        "emptytodoinput":"यू मस्ट रायट समथिंग",
```
Comment (prem-k-r) [inline languages.js:1695]: Also wrong.
Other languages could also be incorrect.
You can leave the translations for languages you're not familiar with.

File: style.css [hunk 4]
```diff
@@ -278,6 +278,9 @@ body {
 	border-radius: 25%;
 	padding: 1px;
 	cursor: pointer;
+}
+
+.theme-transition .dot-icon{
```
Comment (prem-k-r) [inline style.css:283]: Thanks. 

Please do same for ToDo icon

File: languages.js (hunk 2 above)
Comment (Thunder-Blaze) [reply languages.js:882]: ```¯\_(ツ)_/¯``` Used google translate

image

File: languages.js (hunk 3 above)
Comment (Thunder-Blaze) [reply languages.js:1695]: sure I'll leave the translations then 👍

File: style.css (hunk 4 above)
Comment (Thunder-Blaze) [reply style.css:283]: it works for both as I reused the dot-icon class for the to do icon

File: languages.js (hunk 2 above)
Comment (prem-k-r) [reply languages.js:882]: Odd, they transliterated it lol

File: style.css (hunk 4 above)
Comment (prem-k-r) [reply style.css:283]: Ohk nice

File: languages.js (hunk 3 above)
Comment (Thunder-Blaze) [reply languages.js]: done

**Comment by Thunder-Blaze** [comment]
I checked it on Chrome, Brave and Firefox (Windows), so I think it should work properly
but you can check again for cross-platform or cross-browser compatibility issues

**Comment by prem-k-r** [comment]
You removed Hindi too for you must write something by mistake 
Translation for toggle, and To Do list and placeholder type here.. these four are missing 
Rest seems okay

**Comment by Thunder-Blaze** [comment]
I added the translation in English and Hindi
Oh should I change the placeholder to Add task ?

**Comment by prem-k-r** [comment]
> I added the translation for toggle, and To Do list and placeholder in English and Hindi Oh should I change the placeholder to Add task ?

Yeah, would be better, I think

File: languages.js [hunk 5]
```diff
@@ -27,6 +27,8 @@ const translations = {
         "enable_ai_tools": "Show shortcuts for AI tools",
         "googleAppsMenuText": "Google Apps",
         "googleAppsMenuInfo": "Show shortcuts for Google Apps",
+        "todoListText": "To Do List",
+        "todoListInfo": "Show a Daily To Do List",
```
Comment (prem-k-r) [inline languages.js:31]: todoListHeading and todoInput missing

File: languages.js [hunk 6]
```diff
@@ -82,11 +84,13 @@ const translations = {
         "conditionText": "Hello! How are you today?",
         "enterBtn": "Search",
         "searchPlaceholder": "Type here...",
+        "searchPlaceholder": "Type here...",
```
Comment (prem-k-r) [inline languages.js:87]: Extra

File: languages.js [hunk 7]
```diff
@@ -82,11 +84,13 @@ const translations = {
...
         "listenPlaceholder": "Listening...",
-        "searchWithHint": "Search With",
+        "todoPlaceholder": "Add task...",
```
Comment (prem-k-r) [inline languages.js:89]: Changed wrong

File: languages.js (hunk 5 above)
Comment (Thunder-Blaze) [reply languages.js:31]: There was no need to write the same thing twice so I did this

image

Added the todoPlaceholder in my recent commit

File: languages.js (hunk 7 above)
Comment (Thunder-Blaze) [reply languages.js:89]: oh lol, I'll fix this in a min

File: languages.js (hunk 5 above)
Comment (prem-k-r) [reply languages.js:31]: my bad, didn't noticed

File: languages.js (hunk 7 above)
Comment (Thunder-Blaze) [reply languages.js:88]: done 👍

File: languages.js (hunk 5 above)
Comment (Thunder-Blaze) [reply languages.js:31]: no problem :)

File: languages.js (hunk 2 above)
Comment (itz-rj-here) [reply languages.js]: He translation should be "তোমার কিছু লেখা উচিত"

File: languages.js (hunk 5 above)
Comment (itz-rj-here) [reply languages.js:31]: looks like this new person is better than prem 😄

Comment (prem-k-r) [reply languages.js:31]: Hehe.. I always miss something when checking from mobile

Comment (Thunder-Blaze) [reply languages.js:31]: lol no, I am not better
I'm still learning from other great people
Before yesterday I didn't even knew what XSS was
so you have my thanks @prem-k-r

File: languages.js (hunk 2 above)
Comment (Thunder-Blaze) [reply languages.js]: I gave up on languages I don't know.
Someone who knows them can maybe add them later in language.js :)

**Comment by Thunder-Blaze** [comment]
btw @prem-k-r 

image
to
image

I tried fixing the icon but idk if this is right or not ?
If it is right then I'll commit and push the changes

**Comment by prem-k-r** [comment]
Seems like it is.. 
You can make separate PR for this

**Comment by Thunder-Blaze** [comment]
Sure, Roger that 👍🏻

File: languages.js (hunk 5 above)
Comment (itz-rj-here) [reply languages.js:31]: Nah we are just friends. Just making some fun. Thats all. 🙃
Don't worry.

**Comment by Thunder-Blaze** [comment]
btw should I create a new branch and then make a PR
or should I wait for this PR to get accepted and then make the MenuIconFix PR ?

**Comment by prem-k-r** [comment]
> btw should I create a new branch and then make a PR or should I wait for this PR to get accepted and then make the MenuIconFix PR ?

If the addition of code snippets in this PR is not causing the issue, you can create a new branch and raise the PR now without waiting.

**Review by prem-k-r — CHANGES_REQUESTED** [review]
- When a task is marked as completed, the strikethrough data is not being saved; it reverts to a normal task after the page reloads.
- Assigning innerText **directly** from user input (e.g., li.innerText = todoInput.value) is not sanitized. This can still lead to XSS
- It would be better if clicking on to-do icon moves the cursor to the input area.
- Since you are using "\u00d7" (×), if it is contained within a task, it breaks the task into multiple tasks. Also, users can create blank tasks with it, but their row width is shortened.

You may switch to json based approach.. it will solve 2 and 4.

**Comment by Thunder-Blaze** [comment]
https://github.com/user-attachments/assets/aa70c95e-98dc-48df-bed8-70a3b09226cb

I switched to JSON and this is the current situation
I don't know much about XSS (I thought since the h1 tag is getting rendered as normal text, XSS attacks will not work)

- Fixed the problem, now you cannot give empty spaces as input
- Now the strikethrough remains on reload
- Focus shifts to todo input box when clicked on Todo Icon
- You can now use the  "\u00d7" (×) without breaking the todo task into parts

**Comment by prem-k-r** [comment]
we generally use something like this 
function sanitizeInput(input) {
    const div = document.createElement('div');
    div.textContent = input;
    return div.innerHTML;
}

I ask chatgpt, it gave me following code.. please check.. 
there are other minors changes too..
note: there might be mistakes
```
// ----------------------------------- To Do List ----------------------------------------

// DOM elements for managing the To-Do list
const todoContainer = document.getElementById("todoContainer");
const todoListCont = document.getElementById("todoListCont");
const todoulList = document.getElementById("todoullist");
const todoAdd = document.getElementById("todoAdd");
const todoInput = document.getElementById("todoInput");
let todoList = {}; // Object to store To-Do items

// Utility function to sanitize input
function sanitizeInput(input) {
    const div = document.createElement('div');
    div.textContent = input;
    return div.innerHTML;
}

// Add a new To-Do item on button click or "Enter" keypress
todoAdd.addEventListener("click", addtodoItem);
todoInput.addEventListener("keypress", (event) => {
    if (event.key === "Enter") {
        addtodoItem();
    }
});

function addtodoItem() {
    const inputText = todoInput.value.trim();

    // Alert user if input is empty
    if (inputText === '') {
        alert(translations[currentLanguage]?.emptytodoinput || translations['en'].emptytodoinput);
        return;
    }

    const sanitizedText = sanitizeInput(inputText); // Prevent unsafe input

    // Create a new list item with the sanitized input
    let li = document.createElement('li');
    li.textContent = sanitizedText;

    // Add a remove button to the item
    const span = document.createElement("span");
    span.setAttribute("class", "todoremovebtn");
    span.textContent = "\u00d7";

    li.appendChild(span);
    li.addEventListener("click", SetTaskCheckEvent); // Attach event listener for item actions
    li.setAttribute("class", "todolistitem");

    // Generate a unique ID for the item using the current timestamp
    const t = "t" + Date.now();
    li.setAttribute("data-todoitem", t);

    // Store the new item in the todoList object
    todoList[t] = { title: sanitizedText, status: "pending" };
    todoulList.appendChild(li);

    todoInput.value = ''; // Clear the input field
    SaveToDoData(); // Persist the updated list in localStorage
}

// Event handler for marking items as completed or removing them
var SetTaskCheckEvent = (event) => {
    if (event.target.tagName === "LI") {
        // Toggle "checked" status for the item
        event.target.classList.toggle("checked");
        const id = event.target.dataset.todoitem;
        todoList[id].status = todoList[id].status === "completed" ? "pending" : "completed";
        SaveToDoData(); // Update localStorage
    } else if (event.target.tagName === "SPAN") {
        // Remove the item from the list and delete it from the object
        const id = event.target.parentElement.dataset.todoitem;
        event.target.parentElement.remove();
        delete todoList[id];
        SaveToDoData(); // Update localStorage
    }
};

// Save the current state of the To-Do list to localStorage
function SaveToDoData() {
    try {
        localStorage.setItem("todoList", JSON.stringify(todoList));
    } catch (error) {
        console.error("Error saving to localStorage:", error); // Log storage errors
    }
}

// Load and display the To-Do list from localStorage
function ShowToDoList() {
    try {
        todoList = JSON.parse(localStorage.getItem("todoList")) || {}; // Parse stored data or initialize empty

        for (let id in todoList) {
            const todo = todoList[id];

            // Create a list item for each saved To-Do
            let li = document.createElement('li');
            li.textContent = sanitizeInput(todo.title);

            const span = document.createElement("span");
            span.setAttribute("class", "todoremovebtn");
            span.textContent = "\u00d7";

            li.appendChild(span);
            li.addEventListener("click", SetTaskCheckEvent);
            li.setAttribute("class", "todolistitem");

            // Add "checked" class if the item is completed
            if (todo.status === 'completed') {
                li.classList.add("checked");
            }

            li.setAttribute("data-todoitem", id);
            todoulList.appendChild(li);
        }
    } catch (error) {
        console.error("Error loading from localStorage:", error);
        localStorage.setItem("todoList", '{}'); // Reset corrupted data
    }
}

// Initialize the To-Do list based on the date to separate daily tasks
let todoLastUpdateDate = localStorage.getItem("todoLastUpdateDate");
let todoCurrentDate = new Date().toLocaleDateString();

if (todoLastUpdateDate === todoCurrentDate) {
    ShowToDoList(); // Load existing tasks if the date matches
} else {
    // Reset list for a new day
    localStorage.setItem("todoLastUpdateDate", todoCurrentDate);
    localStorage.setItem("todoList", '{}');
    ShowToDoList();
}

// Toggle menu and tooltip visibility
todoListCont.addEventListener("click", function (event) {
    const isMenuVisible = todoContainer.style.display === 'grid';

    // Toggle menu visibility and adjust tooltip
    todoContainer.style.display = isMenuVisible ? 'none' : 'grid';
    if (!isMenuVisible) {
        todoListCont.classList.add('menu-open');
        todoInput.focus(); // Focus input when menu opens
    } else {
        todoListCont.classList.remove('menu-open');
    }
});

// Close menu when clicking outside the To-Do list or menu button
document.addEventListener("click", function (event) {
    const isClickInside = todoContainer.contains(event.target) ||
        todoListCont.contains(event.target) ||
        event.target.classList.contains('todoremovebtn');

    if (!isClickInside && todoContainer.style.display === 'grid') {
        // Hide menu and restore tooltip
        todoContainer.style.display = 'none';
        todoListCont.classList.remove('menu-open');
    }

    event.stopPropagation(); // Prevent event bubbling
});

// ------------------------------- End of To Do List -------------------------------------
```

**Comment by Thunder-Blaze** [comment]
thanks, this helped me learn new js functions
but there is a problem when we are sanitizing the text
image
this becomes this
image
after reloading the page

I think as long as we are getting the data using innerText there should be no XSS vulnerabilities, I even tried the methords of XSS attacks but there were no problems without even sanitizing the input
(I added screenshots in my previous comment)

**Comment by prem-k-r** [comment]
Actually this will most like give warnings in codeql or firefox add-on update

**Comment by prem-k-r** [comment]
>  but there is a problem when we are sanitizing the text image this becomes this image after reloading the page
> 


chatgpt:
```
// Utility function to sanitize input for DOM rendering
function sanitizeInput(input) {
    const div = document.createElement('div');
    div.textContent = input;
    return div.innerHTML;
}

function addtodoItem() {
    const inputText = todoInput.value.trim();

    if (inputText === '') {
        alert(translations[currentLanguage]?.emptytodoinput || translations['en'].emptytodoinput);
        return;
    }

    const t = "t" + Date.now(); // Unique ID
    const rawText = inputText; // Store raw text

    // Store raw text in the todoList object
    todoList[t] = { title: rawText, status: "pending" };

    // Create and append DOM element with sanitized text
    createTodoItemDOM(t, rawText, "pending");

    todoInput.value = ''; // Clear input
    SaveToDoData(); // Save changes
}

function createTodoItemDOM(id, title, status) {
    let li = document.createElement('li');
    li.innerHTML = sanitizeInput(title); // Sanitize before rendering in DOM

    const span = document.createElement("span");
    span.setAttribute("class", "todoremovebtn");
    span.textContent = "\u00d7";

    li.appendChild(span);
    li.addEventListener("click", SetTaskCheckEvent);
    li.setAttribute("class", "todolistitem");

    if (status === 'completed') {
        li.classList.add("checked");
    }

    li.setAttribute("data-todoitem", id);
    todoulList.appendChild(li);
}

function ShowToDoList() {
    try {
        todoList = JSON.parse(localStorage.getItem("todoList")) || {};

        for (let id in todoList) {
            const todo = todoList[id];
            createTodoItemDOM(id, todo.title, todo.status);
        }
    } catch (error) {
        console.error("Error loading from localStorage:", error);
        localStorage.setItem("todoList", '{}');
    }
}
```

**Comment by Thunder-Blaze** [comment]
Done :)

I thought of doing the same but didn't as you said no use of innerHTML
Its working fine now, I used the code you provided 👍🏻

**Comment by Thunder-Blaze** [comment]
ah, I didn't knew what way was direct and what was indirect so don't mind me

and no need to worry about that
image
this line 357 will automatically deal with the problem when there is nothing set in local storage

**Comment by prem-k-r** [comment]
Ok perfect..

**Comment by prem-k-r** [comment]
This is last suggestion lol.
Current code is funtionality and security-wise perfect

Optimized version.. slightly reduces CPU and memory usage
This minimizes reflows for larger lists.

```
// ----------------------------------- To Do List ----------------------------------------

// DOM Variables
const todoContainer = document.getElementById("todoContainer");
const todoListCont = document.getElementById("todoListCont");
const todoulList = document.getElementById("todoullist");
const todoAdd = document.getElementById("todoAdd");
const todoInput = document.getElementById("todoInput");
let todoList = {}; // Initialize todoList JSON

// Add event listeners for Add button click or Enter key press
todoAdd.addEventListener("click", addtodoItem);
todoInput.addEventListener("keypress", (event) => {
    if (event.key === "Enter") {
        addtodoItem();
    }
});

// Utility function to sanitize input
function sanitizeInput(input) {
    const div = document.createElement('div');
    div.textContent = input;
    return div.innerHTML;
}

// Function to add items to the TODO list
function addtodoItem() {
    const inputText = todoInput.value.trim(); // Remove useless whitespaces
    if (inputText === '') {
        alert(translations[currentLanguage]?.emptytodoinput || translations['en'].emptytodoinput);
        return; // Return the function when the input is empty
    }
    const t = "t" + Date.now(); // Generate a Unique ID
    const rawText = inputText;
    todoList[t] = { title: rawText, status: "pending" }; // Add data to the JSON variable

    const li = createTodoItemDOM(t, rawText, "pending"); // Create List item
    todoulList.appendChild(li); // Append the new item to the DOM immediately

    todoInput.value = ''; // Clear Input
    SaveToDoData(); // Save changes
}

function createTodoItemDOM(id, title, status) {
    let li = document.createElement('li');
    li.innerHTML = sanitizeInput(title); // Sanitize before rendering in DOM
    const span = document.createElement("span"); // Create the Cross Icon
    span.setAttribute("class", "todoremovebtn");
    span.textContent = "\u00d7";
    li.appendChild(span); // Add the cross icon to the LI tag
    li.setAttribute("class", "todolistitem");
    if (status === 'completed') {
        li.classList.add("checked");
    }
    li.setAttribute("data-todoitem", id); // Set a data attribute to the li so that we can uniquely identify which li has been modified or deleted
    return li; // Return the created `li` element
}

// Event delegation for task check and remove
todoulList.addEventListener("click", (event) => {
    if (event.target.tagName === "LI") {
        event.target.classList.toggle("checked"); // Check the clicked LI tag
        let id = event.target.dataset.todoitem;
        todoList[id].status = ((todoList[id].status === "completed") ? "pending" : "completed"); // Update status
        SaveToDoData(); // Save Changes
    } else if (event.target.tagName === "SPAN") {
        let id = event.target.parentElement.dataset.todoitem;
        event.target.parentElement.remove(); // Remove the clicked LI tag
        delete todoList[id]; // Remove the deleted List item data
        SaveToDoData(); // Save Changes
    }
});

// Save JSON to local Storage
function SaveToDoData() {
    localStorage.setItem("todoList", JSON.stringify(todoList));
}

// Fetch saved JSON and create list items using it
function ShowToDoList() {
    try {
        todoList = JSON.parse(localStorage.getItem("todoList")) || {}; // Parse stored data or initialize empty
        const fragment = document.createDocumentFragment(); // Create a DocumentFragment
        for (let id in todoList) {
            const todo = todoList[id];
            const li = createTodoItemDOM(id, todo.title, todo.status); // Create `li` elements
            fragment.appendChild(li); // Add `li` to the fragment
        }
        todoulList.appendChild(fragment); // Append all `li` to the `ul` at once
    } catch (error) {
        console.error("Error loading from localStorage:", error);
        localStorage.setItem("todoList", '{}'); // Reset corrupted data
    }
}

// Code to reset the List on the Next Day
let todoLastUpdateDate = localStorage.getItem("todoLastUpdateDate"); // Get the date of last update
let todoCurrentDate = new Date().toLocaleDateString(); // Get current date
if (todoLastUpdateDate === todoCurrentDate) {
    ShowToDoList();
} else {
    // Reset the list when last update date and the current date does not match
    localStorage.setItem("todoLastUpdateDate", todoCurrentDate);
    localStorage.setItem("todoList", '{}');
    ShowToDoList();
}

// Toggle menu and tooltip visibility
todoListCont.addEventListener("click", function (event) {
    const isMenuVisible = todoContainer.style.display === 'grid';

    // Toggle menu visibility
    todoContainer.style.display = isMenuVisible ? 'none' : 'grid';

    // Add or remove the class to hide the tooltip
    if (!isMenuVisible) {
        todoListCont.classList.add('menu-open'); // Hide tooltip
        todoInput.focus();
    } else {
        todoListCont.classList.remove('menu-open'); // Restore tooltip
    }
});

// Close menu when clicking outside
document.addEventListener("click", function (event) {
    const isClickInside =
        todoContainer.contains(event.target) || todoListCont.contains(event.target) || event.target.classList.contains('todoremovebtn');

    if (!isClickInside && todoContainer.style.display === 'grid') {
        todoContainer.style.display = 'none'; // Hide menu
        todoListCont.classList.remove('menu-open'); // Restore tooltip
    }

    event.stopPropagation();
});

// ------------------------------- End of To Do List -------------------------------------
```

**Review by prem-k-r — APPROVED** [review]

**Comment by prem-k-r** [comment]
@XengShi good to go, i think
You can review and merge it

**Review by itz-rj-here — APPROVED** [review]
The codes look fine to me :)
Later i will check on my computer.

**Comment by XengShi** [comment]
Great work, but you can remove this :
image

Just do nothing when clicking on "+" button if there is empty string

**Comment by Thunder-Blaze** [comment]
thanks :) 
Done 👍🏻

**Comment by Thunder-Blaze** [comment]
Wait, I'll try to fix that
It shouldn't be much hard

**Comment by Thunder-Blaze** [comment]
And done, should I make a seperate PR for that ?

**Comment by prem-k-r** [comment]
Yeah

**Pull request merged**